lib/item.rb

```diff
@@ -0,0 +1,59 @@
+require 'httparty'
+require "pry"
+
+class Item
+  attr_reader :nypl_source, :id
+  attr_accessor :item_type_code, :location_code
+
+  def initialize(nypl_source, id)
+    @nypl_source = nypl_source
+    @id = id
+  end
+
+  def is_research?
+    get_platform_api_data
+
+    result = is_partner? || item_type_is_research? || location_is_only_research?
+
+    log_data = {is_partner: is_partner?}
+
+    if !is_partner?
+      log_data[:item_type_is_research] = item_type_is_research?
+      log_data[:location_is_only_research] = location_is_only_research?
+    end
+
+    $logger.debug "Evaluating is-research for #{nypl_source} #{id}: #{result}", log_data
```

Comment:
You are only really evaluating for `location_is_only_research?` if `item_type_is_research?` is falsey. I would probably make lines 21 and 22 (where `log_data' gets updated) part of the `item_type_is_research?` and `location_is_only_research?` methods, and same with the `$logger.debug` statement on line 25 (which would become two debug statements, one in each message). That would clean up this method and put the logging/updating in the methods that it is attached to. `log_data` could either become an instance variable or an argument of these two methods in order for that to work.